src/sink/wavefront.rs

```diff
@@ -317,43 +352,24 @@ impl Sink for Wavefront {
             if self.delivery_attempts > 0 {
                 debug!("delivery attempts: {}", self.delivery_attempts);
             }
-            let addrs = (self.host.as_str(), self.port).to_socket_addrs();
-            match addrs {
-                Ok(srv) => {
-                    let ips: Vec<_> = srv.collect();
-                    for ip in ips {
-                        match TcpStream::connect(ip) {
-                            Ok(mut stream) => {
-                                let res = stream.write(self.stats.as_bytes());
-                                if res.is_ok() {
-                                    self.stats.clear();
-                                    self.delivery_attempts = 0;
-                                    return;
-                                } else {
-                                    self.delivery_attempts =
-                                        self.delivery_attempts.saturating_add(1);
-                                }
-                            }
-                            Err(e) => {
-                                info!(
-                                    "Unable to connect to proxy at {} using addr {} with error \
-                                     {}",
-                                    self.host,
-                                    ip,
-                                    e
-                                )
-                            }
-                        }
-                        time::delay(self.delivery_attempts);
-                    }
-                }
-                Err(e) => {
-                    info!(
-                        "Unable to perform DNS lookup on host {} with error {}",
-                        self.host,
-                        e
-                    );
+            let mut delivery_failure = false;
+            if let Some(ref mut stream) = self.stream {
+                let res = stream.write_all(self.stats.as_bytes());
+                if res.is_ok() {
+                    self.aggrs.reset();
+                    self.stats.clear();
+                    self.delivery_attempts = 0;
+                    return;
+                } else {
+                    self.delivery_attempts = self.delivery_attempts.saturating_add(1);
+                    delivery_failure = true;
                 }
+            } else {
+                time::delay(self.delivery_attempts);
```

Comment:
`time::delay` comes from [here](https://github.com/postmates/cernan/blob/master/src/time.rs#L38). The inputs are linear but the backoff occurs as the power of two, up to a cap. We special-case the `0` to be a no-op https://github.com/postmates/cernan/blob/master/src/time.rs#L40.